src/widgets/scrollbar.rs

```diff
@@ -446,198 +449,177 @@ impl<'a> Scrollbar<'a> {
+    ///   │               █
+    ///   │               █<──────── thumb_start
+    ///   │               ║<──────── track_end
+    ///   └───────────────┘
+    /// ```
     ///
-    /// This function returns a tuple `(thumb_start, thumb_end)` where `thumb_start` is the position
-    /// at which the scrollbar thumb begins, and `thumb_end` is the position at which the
-    /// scrollbar thumb ends.
+    /// For ScrollbarOrientation::HorizontalBottom
+    ///
+    /// ```plain
+    ///   ┌───────────────┐
+    ///   │               │
+    ///   │               │
+    ///   │               │
+    ///   └═══█████═══════┘
+    ///    ^  ^   ^      ^
+    ///    │  │   │      └────────── track_end
+    ///    │  │   │
+    ///    │  │   └───────────────── thumb_end
+    ///    │  │
+    ///    │  └───────────────────── thumb_start
+    ///    │
+    ///    └──────────────────────── track_start
+    /// ```
     ///
-    /// The size of the thumb (i.e., `thumb_end - thumb_start`) is proportional to the ratio of the
-    /// viewport content length to the total content length.
+    /// Specifically this function returns the lengths of the different segments:
     ///
-    /// The position of the thumb (i.e., `thumb_start`) is proportional to the ratio of the current
-    /// scroll position to the total content length.
-    fn get_thumb_start_end(
-        &self,
-        state: &ScrollbarState,
-        track_start_end: (u16, u16),
-    ) -> (u16, u16) {
-        // let (track_start, track_end) = track_start_end;
-        // let track_size = track_end - track_start;
-        // let thumb_size =
-        //     ((state.viewport_content_length / state.content_length) * track_size).max(1);
-        // let thumb_start = (state.position / state.content_length) *
-        //                    state.viewport_content_length;
-        // let thumb_end = thumb_size + thumb_start;
-        // (thumb_start, thumb_end)
-
-        let (track_start, track_end) = track_start_end;
-
-        let viewport_content_length = if state.viewport_content_length == 0 {
-            (track_end - track_start) as usize
-        } else {
-            state.viewport_content_length
+    /// ```plain
+    ///         ┌──────────── thumb_len
+    ///       vvvvv
+    ///    ═══█████═══════
+    ///    ^^^     ^^^^^^^
+    ///     │         └────── track_end_len
+    ///     │
+    ///     └──────────────── track_start_len
+    /// ```
+    fn get_track_lens(&self, area: Rect, state: &mut ScrollbarState) -> (usize, usize, usize) {
+        let (mut track_start, mut track_end) = match self.orientation {
+            ScrollbarOrientation::VerticalRight => (area.y, (area.y + area.height)),
+            ScrollbarOrientation::VerticalLeft => (area.y, (area.y + area.height)),
+            ScrollbarOrientation::HorizontalBottom => (area.x, (area.x + area.width)),
+            ScrollbarOrientation::HorizontalTop => (area.x, (area.x + area.width)),
         };
```

Comment:
Pass this value into the method (or calculate it by calling some other small method). Just deal in calculating the lengths not the direction.